src/win/pipe.c

```diff
@@ -942,20 +951,81 @@ int uv_pipe_listen(uv_pipe_t* handle, int backlog, uv_connection_cb cb) {
 }
 
 
+static DWORD WINAPI uv__pipe_zero_read_interrupter(void* param) {
+  uv__pipe_interrupter_param_t* interrupterParam;
+  int r;
+
+  interrupterParam = (uv__pipe_interrupter_param_t*) param;
+  for (;;) {
+    r = WaitForSingleObject(interrupterParam->completedEvent,
+                            PIPE_ZERO_READ_INTERRUPT_INTERVAL);
+    if (r == WAIT_TIMEOUT) {
```

Comment:
this is still race-y. getting a wait-timeout previous to here doesn't guarantee that the thread handle is still valid after here.